CMakeLists.txt

```diff
@@ -109,16 +109,28 @@ else()
 
   # No config file for this
   if(WITH_GFLAGS)
-    find_package(gflags REQUIRED)
+    option(GFLAGS_USE_TARGET_NAMESPACE "Use gflags import target with namespace." ON)
+    find_package(gflags CONFIG)
+    if(gflags_FOUND)
+      list(APPEND THIRDPARTY_LIBS ${GFLAGS_TARGET})
+    else()
+      find_package(gflags REQUIRED)
+      list(APPEND THIRDPARTY_LIBS gflags::gflags)
+      include_directories(${gflags_INCLUDE_DIR})
+    endif()
     add_definitions(-DGFLAGS=1)
-    include_directories(${gflags_INCLUDE_DIR})
-    list(APPEND THIRDPARTY_LIBS gflags::gflags)
   endif()
 
   if(WITH_SNAPPY)
-    find_package(snappy REQUIRED)
+    find_package(Snappy CONFIG)
```

Comment:
@xkszltl since you are here, could you make Snappy `REQUIRED` here? as `WITH_SNAPPY` is OFF by default, so if it's enabled because it is explicitly requested by user, if it's not found, we should bail out and fail.

Reply:
No. We should allow CONFIG to failed if missing, but fallback path (general lookup, i.e. try MODULE then CONFIG) should have REQUIRED as final guard.